model/webhook.go

```diff
@@ -0,0 +1,162 @@
+package model
+
+import (
+	"encoding/json"
+	"github.com/buger/jsonparser"
+)
+
+type WebhookEntities struct {
+	WebhookEntities []WebhookEntity `json:"Entities"`
+}
+
+type WebhookEntity struct {
+	Name   string              `json:"Name"`
+	Events WebhookEntityEvents `json:"Events"`
+}
+
+func (w *WebhookEntities) UnmarshalJSON(b []byte) error {
+	rawWebhooks, dataType, _, err := jsonparser.Get(b, "Entities", "Entity")
+	if err != nil && err != jsonparser.KeyPathNotFoundError {
+		return err
+	}
+
+	if len(rawWebhooks) == 0 || dataType == jsonparser.NotExist {
+		*w = WebhookEntities{[]WebhookEntity{}}
+		return nil
+	}
+
+	var webhookEntities []WebhookEntity
+	switch dataType {
+	case jsonparser.Array:
+		if err := json.Unmarshal(rawWebhooks, &webhookEntities); nil != err {
+			return err
+		}
+	case jsonparser.Object:
+		var webhookEntity WebhookEntity
+		if err := json.Unmarshal(rawWebhooks, &webhookEntity); nil != err {
+			return err
+		}
+
+		webhookEntities = []WebhookEntity{webhookEntity}
+	}
+
+	*w = WebhookEntities{webhookEntities}
+
+	return nil
+}
+
+type WebhookEntityEvents struct {
+	Events []WebhookEntityEvent `json:"Events"`
+}
+
+type WebhookEntityEvent struct {
+	EventName  string `json:"EventName"`
+	EventAlias string `json:"EventAlias"`
+}
+
+func (w *WebhookEntityEvents) UnmarshalJSON(b []byte) error {
+	rawEvents, dataType, _, err := jsonparser.Get(b, "Event")
```

Comment:
I'm not sure. shouldn't it be `jsonparser.Get(b, "Events", "Event")`?

Reply:
no, what we have in `b` is `{"Event":...}`